apps/web/src/components/Solana/WalletContextProvider.tsx

```diff
@@ -0,0 +1,58 @@
+"use client";
+
+import { BackpackWalletAdapter } from "@solana/wallet-adapter-backpack";
+import type { Adapter, WalletError } from "@solana/wallet-adapter-base";
+import { WalletAdapterNetwork } from "@solana/wallet-adapter-base";
+import { PhantomWalletAdapter } from "@solana/wallet-adapter-phantom";
+import {
+  ConnectionProvider,
+  WalletProvider,
+} from "@solana/wallet-adapter-react";
+import { SolflareWalletAdapter } from "@solana/wallet-adapter-solflare";
+import { clusterApiUrl } from "@solana/web3.js";
+import { type FC, type ReactNode, useCallback, useMemo } from "react";
+import { WalletModalProvider } from "./WalletModalProvider";
+
+export const WalletContextProvider: FC<{ children: ReactNode }> = ({
```

Comment:
We should talk about this, as we will need to incorporate some patterns to allow this provider to not just make the entire app a client component! I.e. using it server side, or if needed on client, only around the components that need it.